pkg/cinderbackup/statefulset.go

```diff
@@ -123,12 +132,21 @@ func StatefulSet(
 								RunAsUser:  &runAsUser,
 								Privileged: &trueVar,
 							},
-							Env:            env.MergeEnvs([]corev1.EnvVar{}, envVars),
-							VolumeMounts:   GetVolumeMounts(),
-							Resources:      instance.Spec.Resources,
-							ReadinessProbe: readinessProbe,
-							LivenessProbe:  livenessProbe,
-							StartupProbe:   startupProbe,
+							Env:           finalEnvVars,
+							VolumeMounts:  volumeMounts,
+							Resources:     instance.Spec.Resources,
+							LivenessProbe: livenessProbe,
+							StartupProbe:  startupProbe,
+						},
+						{
+							Name:    "probe",
+							Command: probeCommand,
+							Image:   instance.Spec.ContainerImage,
+							SecurityContext: &corev1.SecurityContext{
+								RunAsUser: &runAsUser,
```

Comment:
Could this run as the cinder user?

Reply:
I can run it, but currently cinder user is not 1000, it is 42407, same as its group.
I feared that setting this magic number may break us in the future since the container may change this at any point, but I dig a bit more after your comment and based on [where the magic number comes from](https://github.com/openstack/kolla/blob/0d4e8e6c11947e07953121d6009809db69f4965b/kolla/common/users.py#L40-L42) I don't think we have much to fear so I'll update the patch.